internal/pkg/service/common/etcdop/iterator/iterator_typed.go

```diff
@@ -38,7 +39,37 @@ func (v DefinitionT[T]) Do(ctx context.Context, opts ...op.Option) *IteratorT[T]
 
 // ForEachOp method converts iterator to for each operation definition, so it can be part of a transaction.
 func (v DefinitionT[T]) ForEachOp(fn func(value T, header *Header) error) *ForEachOpT[T] {
-	return &ForEachOpT[T]{def: v, fn: fn}
+	return &ForEachOpT[T]{def: v, onValue: fn}
+}
+
+// WithResultTo method converts iterator to for each operation definition, so it can be part of a transaction.
+func (v DefinitionT[T]) WithResultTo(slice *[]T) *ForEachOpT[T] {
+	return v.
+		ForEachOp(func(value T, header *Header) error {
+			*slice = append(*slice, value)
+			return nil
+		}).
+		AndOnFirstPage(func(response *etcd.GetResponse) error {
+			*slice = nil
+			return nil
+		})
+}
+
+// AndOnFirstPage registers a callback that is executed after the first page is successfully loaded.
+func (v *ForEachOpT[T]) AndOnFirstPage(fn func(response *etcd.GetResponse) error) *ForEachOpT[T] {
+	return v.AndOnPage(func(pageIndex int, response *etcd.GetResponse) error {
+		if pageIndex == 0 {
+			return fn(response)
+		}
+		return nil
+	})
+}
+
+// AndOnPage registers a callback that is executed after each page is successfully loaded.
+func (v *ForEachOpT[T]) AndOnPage(fn onPageFn) *ForEachOpT[T] {
+	clone := *v
+	clone.onPage = append(clone.onPage, fn)
+	return &clone
 }
 
 func (v *ForEachOpT[T]) Op(ctx context.Context) (op.LowLevelOp, error) {
```

Comment:
`Op` can be used to include the operation in `TxnOp`/`AtomicOp`.

`Do` bellow can be use to perform the operation directly.